projects/ad469x_iio/app/ad469x_iio.c

```diff
@@ -1129,17 +1134,30 @@ static int32_t ad469x_iio_submit_samples(struct iio_device_data *iio_dev_data)
 	}
 #else // SPI_DMA_MODE
 #if (DATA_CAPTURE_MODE == BURST_DATA_CAPTURE)
+	nb_of_samples *= BYTES_PER_SAMPLE;
 	ret = no_os_cb_prepare_async_write(iio_dev_data->buffer->buf,
-					   nb_of_samples * (BYTES_PER_SAMPLE), &buff_start_addr, &data_read);
+					   nb_of_samples, &buff_start_addr, &data_read);
 	if (ret) {
 		return ret;
 	}
 
 	if (!dma_config_updated) {
```

Comment:
In software , explicitly checking the condition is good practice. something like 
if (a==0) instead of if(!a).. I